server/latest.go

```diff
@@ -0,0 +1,258 @@
+	s3Prefix := ver + "/"
+	if typeToRelease == "desktop" {
+		s3Prefix = "desktop/" + ver + "/"
+	}
+	input := &s3.ListObjectsV2Input{
+		Bucket: aws.String(releaseBucket),
+		Prefix: aws.String(s3Prefix),
+	}
+
+	LogInfo("Checking for s3 bucket: %s", releaseBucket+"/"+s3Prefix)
+	result, _ := checkBucket(svc, input)
+	if *result.KeyCount != int64(0) {
+		return result, nil
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			LogError("Timed out waiting for %s to be created", releaseBucket+"/"+s3Prefix)
+			return nil, ctx.Err()
+		case <-time.After(5 * time.Minute):
+			result, _ = checkBucket(svc, input)
+			if *result.KeyCount != int64(0) {
+				return result, nil
+			}
+			timeLeft, _ := ctx.Deadline()
+			LogInfo("Release %s is not available; waiting until: %s", releaseBucket+"/"+s3Prefix, timeLeft.Format("15:04:05"))
+		}
+	}
+}
+
+func preserverExistingRoutingRules(svc *s3.S3, cfg *MatterbuildConfig, typeToRelease string, params s3.PutBucketWebsiteInput) error {
+
+	bucketConfig, err := getBucketConfig(svc, cfg.S3BucketNameForLatestURLs)
+	if err != nil {
+		LogError("Unable to get the %s AWS Bucket Website Config.", cfg.S3BucketNameForLatestURLs)
+		return err
+	}
+
+	// Check for Routing Rules that are not related to the typeToRelease value and carry them forward
+	for _, value := range bucketConfig.RoutingRules {
+		checked := false
+
+		if typeToRelease == "desktop" {
+			checked = (strings.Contains(*value.Condition.KeyPrefixEquals, "enterprise") || strings.Contains(*value.Condition.KeyPrefixEquals, "team"))
+		}
+
+		if typeToRelease == "server" {
+			checked = strings.Contains(*value.Condition.KeyPrefixEquals, "desktop")
+		}
+
+		if checked {
+			params.WebsiteConfiguration.RoutingRules = append(params.WebsiteConfiguration.RoutingRules, value)
+			LogInfo("Copying rule %s forward as it is not being updated", *value.Condition.KeyPrefixEquals)
+		}
+	}
+
+	return nil
+}
+
+func addRoutingRule(file string, keyToUse string, params s3.PutBucketWebsiteInput, suffix string) error {
+	valueToAdd := &s3.RoutingRule{
+		Condition: &s3.Condition{
+			KeyPrefixEquals: aws.String(keyToUse + suffix),
+		},
+		Redirect: &s3.Redirect{
+			Protocol:       aws.String("https"),
+			HostName:       aws.String("releases.mattermost.com"),
+			ReplaceKeyWith: aws.String(file),
+		},
+	}
+	LogInfo("Adding %s RoutingRule for: %s", keyToUse+suffix, file)
+	params.WebsiteConfiguration.RoutingRules = append(params.WebsiteConfiguration.RoutingRules, valueToAdd)
+
+	return nil
+}
+
+func generateNewRoutesForRelease(cfg *MatterbuildConfig, result *s3.ListObjectsV2Output, fileSearchValue string, ver string, params s3.PutBucketWebsiteInput) error {
+
+	for _, value := range result.Contents {
+		if strings.Contains(*value.Key, fileSearchValue) && !strings.Contains(*value.Key, ".sig") {
+			switchValue := *value.Key
+			switch {
+			case strings.HasSuffix(switchValue, ".dmg"):
+				addRoutingRule(*value.Key, fileSearchValue, params, "-dmg")
```

Comment:
```suggestion
				_ = addRoutingRule(*value.Key, fileSearchValue, params, "-dmg")
```
Could we change it to this please?